src/psyclone/tests/core/component_indices_test.py

```diff
@@ -102,3 +102,18 @@ def test_iterating():
     for count, indx in enumerate(component_indices.iterate()):
         assert correct_index_pairs[count] == indx
         assert correct[count] == component_indices[indx]
+
+
+# -----------------------------------------------------------------------------
+def test_component_indices_getitem_exceptions():
+    '''Tests useful error messages are provides if a tuple is provided that
+    is out of bounds.
+    '''
+    component_indices = ComponentIndices([["a", "b"], ["c"]])
+    with pytest.raises(IndexError) as err:
```

Comment:
Please could you extend this to cover the lower-bound cases too (e.g. attempting to access `-1`).

Reply:
Done.